app/src/main/java/com/kickstarter/libs/Either.kt

```diff
@@ -0,0 +1,63 @@
+package com.kickstarter.libs
+
+class Either<out A, out B> private constructor (val left: A?, val right: B?) {
```

Comment:
What do you think of using sealed classes for this?

``` kotlin
sealed class Either<out L, out R> {
  class Left<out L, out R>(val left: L) : Either<L, R> {
    ...
  }
  class Right<out L, out R>(val right: R) : Either<L, R> {
    ...
  }
}

Either.Left("Error!")
Either.Right(1)

when(either) {
  is Left -> either.left
  is Right -> either.right
}
```

Seems to be a common pattern for tagged unions in Kotlin, and prevents even the private ability of instantiating an invalid type.